src/goto-programs/goto_check.cpp

```diff
@@ -394,6 +394,9 @@ void goto_checkt::shift_check(
   if(!enable_ub_shift_check)
     return;
 
+  if(!is_lshr2t(expr) && !is_ashr2t(expr) && !is_shl2t(expr))
```

Comment:
Why are we arriving at this function with anything besides a shift? Shouldn't we:

- Add an assertion that this function is only reachable within shift operations
- Fix the flow that is calling a shift check on something that it shouldn't

Reply:
>Why are we arriving at this function with anything besides a shift

We didn't have a break in the switch-case of goto_checkt::check_rec(...) after case expr2t::modulus_id, so shift_check was being called on this operation. Initially, I preferred a minimal change to reduce the chances of side effects, but I agree with you that it is better to adjust the flow, and it seems safe to modify it. I will update the commit. Thank you for the suggestion.